janus_server/src/aggregator/accumulator.rs

```diff
@@ -103,11 +103,11 @@ where
     /// aggregation exists, one is created and initialized with the accumulated values.
     #[tracing::instrument(skip(self, tx), err)]
     pub(super) async fn flush_to_datastore<C: Clock>(
-        self,
+        &self,
```

Comment:
I think I deliberately made this method consume `self` so that you can't continue accumulating into a flushed accumulator. Does making it take a reference make something else more ergonomic?

Reply:
It's required for use in the leader aggregation implementation: since the computations around what to write (i.e. `update` calls) happen outside a transaction, but the writes (i.e. `flush_to_datastore`) happen inside a transaction, and transactions can be forced to retry due to serialization issues, the `flush_to_datastore` operation can't consume the accumulator since we might need to try writing again.

(We might need this in the helper aggregation implementation eventually too, if/when we break all of the aggregation computations out of the datastore transaction.)

We could replace this with a runtime check if we wanted to be extra-careful, but I am OK with code inspection + tests here to make sure we're doing the right thing.

Reply:
Sounds reasonable.